### PR TITLE
Congrat employees on anniversary of working in the company

### LINKED ISSUE #142: Congrat employees on anniversary of working in the company

### PR DESCRIPTION
**Pull request by Hellikrox**
Closed #142 
# Please make sure all below checkboxes in the Mandatory section have been checked before submitting your PR (also don't forget to pay attention to the Depending on Circumstances section)

## Mandatory

- [x] The commit message is an imperative and starts with a capital letter (for example, `Add something`, `Remove something`, `Fix something`, etc.).
- [x] I made sure I hadn't put a period at the end of the the commit message(s).

## Depending on Circumstances

Some of the items in the section become mandatory if you are a first-time contributor and/or your changes are relevant to the items.

- [ ] Adding a new parameter I didn't forget to reflect it in `README.md`.
- [x] Modifying a list of constants or imports I didn't brake an alphabetical order.
- [x] (for first-time contributors) One of the commits in the pull request adds me to the **Acknowledgements** section in `AUTHORS.md`.

### CODE REVIEW

File: src/birthder.js [hunk 1]
```diff
@@ -636,7 +690,12 @@
 
     // Check regularly if today is someone's birthday, write birthday messages to the general channel.
     if (HAPPY_REMINDER_SCHEDULER) {
-      schedule.scheduleJob(HAPPY_REMINDER_SCHEDULER, () => sendCongratulations(robot))
+      schedule.scheduleJob(HAPPY_REMINDER_SCHEDULER, () => sendCongratulations(robot, 'bday'))
+    }
+
+    // Check regularly if today is someone's anniversary day, write congratulation message to the general channel.
+    if (HAPPY_REMINDER_SCHEDULER) {
+      schedule.scheduleJob(HAPPY_REMINDER_SCHEDULER, () => sendCongratulations(robot, 'fwd'))
```
Comment (eugulixes) [inline src/birthder.js]: Let's use a constant named `FWD_EVENT_TYPE` for `fwd`.

File: src/birthder.js [hunk 2]
```diff
@@ -636,7 +690,12 @@
 
     // Check regularly if today is someone's birthday, write birthday messages to the general channel.
     if (HAPPY_REMINDER_SCHEDULER) {
-      schedule.scheduleJob(HAPPY_REMINDER_SCHEDULER, () => sendCongratulations(robot))
+      schedule.scheduleJob(HAPPY_REMINDER_SCHEDULER, () => sendCongratulations(robot, 'bday'))
```
Comment (eugulixes) [inline src/birthder.js]: Let's use a constant named `BDAY_EVENT_TYPE` for `bday`.

File: README.md [hunk 3]
```diff
@@ -10,14 +10,19 @@ Hubot script for writing birthday messages to users. It uses [Tenor](https://ten
...
+  * Writing birthday messages to `#general`.
+  * Reminding users of the upcoming birthdays one day and a few days in advance. It's possible to specify how long before the event occurs the reminder should be triggered.
+  * Providing fault tolerance:
+    + if Tenor is for some reason unavailable right now, the script will try to request it a number of times with delays;
+    + if all the requests failed, users will receive their birthday messages anyway.
+  * Creating private channels in advance and inviting all the users to the channels except the birthday boys/girls for the purpose of discussing the presents for them. Expired channels are automatically removed.
+- Work anniversaries
+  * Memorizing the date when a new employee logs in to Rocket.Chat for the first time (the date is equal to the first working day in the company).
```
Comment (polina-popova) [inline README.md:24]:
```
... when a new employee was added to `general` ...
```

File: README.md [hunk 4]
```diff
@@ -10,14 +10,19 @@ Hubot script for writing birthday messages to users. It uses [Tenor](https://ten
...
+  * Congratulating employees on anniversary of working in the company.
```
Comment (polina-popova) [inline README.md]: For birthday we specify the channel for sending congratulations, probably we should do it here too.

File: README.md (hunk 3 above)
Comment (eugulixes) [reply README.md:24]: It's a detail of implementation, so I guess we shouldn't mention here anything related to the implementation.

File: src/utils.js [hunk 5]
```diff
@@ -47,19 +51,47 @@ exp.sorting = (a, b, format) => {
...
-  if (BIRTHDAY_CHANNEL_MESSAGE_INDEX >= exp.BIRTHDAY_CHANNEL_MESSAGE.length) {
-    BIRTHDAY_CHANNEL_MESSAGE_INDEX = 0
+  if (event === exp.BDAY_EVENT_TYPE) {
+    fileName = '/quotes.txt'
+    index = BIRTHDAY_CHANNEL_MESSAGE_INDEX
   }
 
-  return result
+  const quotesPath = path.join(__dirname, fileName)
+  let quotes = fs.readFileSync(quotesPath, 'utf8').toString().split('\n')
```
Comment (ya7on) [inline src/utils.js]: `fs.readFileSync` is already returns String type. There is no need to make it string

File: src/utils.js [hunk 6]
```diff
@@ -47,19 +51,47 @@ exp.sorting = (a, b, format) => {
...
+      quotes = exp.BIRTHDAY_CHANNEL_MESSAGE
+    }
+    const result = quotes[index]
+
+    index++
+    if (index >= quotes.length) {
+      index = 0
+    }
+
+    event === exp.FWD_EVENT_TYPE ? FWD_MESSAGE_INDEX = index : BIRTHDAY_CHANNEL_MESSAGE_INDEX = index
```
Comment (ya7on) [inline src/utils.js]: It is not very beautiful. It is bad to use the ternary operator for different variables.

**Review by ya7on — DISMISSED** [review]
It's work

**Review by ya7on — DISMISSED** [review]

**Review by ya7on — APPROVED** [review]

**Pull request merged**